src/coreclr/vm/peimagelayout.cpp

```diff
@@ -725,6 +731,11 @@ FlatImageLayout::FlatImageLayout(PEImage* pOwner)
             ThrowHR(E_FAIL); // we don't have any indication of what kind of failure. Possibly a corrupt image.
 #endif
         }
+#else // !TARGET_ANDROID
+        m_FileMap.Assign(mapBegin);
+        m_FileView.Assign(mapBegin);
```

Comment:
m_FileMap and m_FileView have destructors that will try to free what they are pointing to as a file-mapping / as HANDLE.

Assigning a pointer that's not a file-mapping / a HANDLE is going to lead to inscrutable crashes when the destructor runs.

Reply:
This is done this way to make the PR work, I think the correct way should be implemented in the follow up. Android applications don't quit unless they are killed, so **for now**, I think it's acceptable.

Reply:
I am not comfortable with merging code to main that's obviously wrong. (I am ok with merging code that's not finished.)